pages/docs/tutorials/websocket/index.md

```diff
@@ -0,0 +1,343 @@
+
+
+```mermaid
+sequenceDiagram
+Client -->> Server: Initiate WebSocket Connection
+Server -->> Client: Message 1
+Server -->> Client: Message 2
+Server -->> Client: Message 3
+Server -->> Client: Message 4
+```
+
+In Slack, WebSocket is employed as part of its [Socket Mode](https://api.slack.com/apis/connections/socket) feature to facilitate real-time notifications between Slack's servers and third-party applications or bots. The [Slack Event API](https://api.slack.com/apis/connections/events-api) is a tool that lets you receive real-time notifications of specific events in a Slack workspace such as messages, reactions, and user presence changes.
+
+
+## Define AsyncAPI version, API information, and server
+
+You start your AsyncAPI document by specifying the AsyncAPI version and essential information about your Slack application's API which includes details such as the `title`, `version` and `description`.
+
+The `servers` section allows you to define the protocol and specify information about the URLs your application will use, such as  `host`, `pathname`, `protocol` and `description`.
+
+<Remember>
+The WebSocket URL is generated  by invoking the <a href="https://api.slack.com/methods/apps.connections.open">apps.connections.open</a> method from Slack’s API. You use the authentication tokens obtained during the configuration of your Slackbot to generate this URL.
+</Remember>
+
+```
+asyncapi: '3.0.0'
+info:
+  title: Create an AsyncAPI Document for a Slackbot with WebSockets
+  version: '1.0.0'
+  description:  |
+    The Heart-Counter manages popular messages in a Slack workspace by monitoring message reaction data
+    
+servers:
+  production:
+    host: wss-primary.slack.com
+    pathname: /link
+    protocol: wss
+    description: Slack's server in Socket Mode for real-time communication 
+```
+
+## Define messages and schemas
+
+Your AsyncAPI document needs to be very clear on the type of event it is expected to receive. Here's where the `messages` component steps in. Using the `payload` property, you can specify what these events should look like, their structure, and what content they carry.
+
+The `payload` attribute specifies the name, format, and description of all expected properties, and can even set constant values that must be followed during schema validation.
+For example, in `reaction` schema definition, any API message received from this channel must follow the constant value for the `reaction` property which is clearly defined as “heart”.
+
+The `const` value feature ensures that the data exchanged through your API complies with your specified constants, helping to maintain data integrity and accuracy.
```

Comment:
unfortunately you will have to remove this `const` part for `reaction` schema that you added to enforce acceptance of `heart` reactions only. Thing is that now we know about https://api.slack.com/apis/connections/socket#acknowledge which means that you cannot by default reject some specific reaction events basing on `reaction` string, as you need to acknowledge every single event